### PR TITLE
Added preview build mode.

### PR DESCRIPTION
**Pull request by feldergast**
Removed --enable-deprecated from configure and add --enable-preview-build.  Used the new flag to remove all the code that is targeted to be removed in SST 10.0.

### CODE REVIEW

File: config/sst_check_deprecated.m4 [hunk 1]
```diff
@@ -1,21 +0,0 @@
-AC_DEFUN([SST_CHECK_DEPRECATED],
-[
-  sst_check_hdf5_happy="yes"
-
-  AC_ARG_ENABLE([deprecated],
-    [AS_HELP_STRING([--(dis|en)able-deprecated],
-      [Enable or disable deprecated code (Default: enabled)])],
-    [
```
Comment (jjwilke) [inline config/sst_check_deprecated.m4]: Did you forget to add a file? Or am I just misreading? I don't see where the `CHECK_PREVIEW_BUILD` function is defined.

Comment (feldergast) [reply config/sst_check_deprecated.m4]: Yes.  Good catch.  File added.  I also pushed some other changes to get rid of the Component* version SubComponent constructor as a valid API.

File: src/sst/core/baseComponent.cc [hunk 2]
```diff
@@ -312,13 +319,15 @@ BaseComponent::configureLink(const std::string& name, TimeConverter* time_base,
                     my_info->link_map = myLinks;
                 }
                 myLinks->insertLink(name,tmp);
+#ifndef SST_ENABLE_PREVIEW_BUILD
                 // Need to set the link's defaultTimeBase to nullptr,
                 // except in the case of this being an Anonymously
                 // loadeed SubComponent, then for backward
                 // compatibility, we leave it as is.
                 if ( !my_info->isLegacySubComponent() ) {
                     tmp->setDefaultTimeBase(nullptr);
```
Comment (gvoskuilen) [inline src/sst/core/baseComponent.cc]: Should defaulttimebase be set to nullptr for all subcomponents if SST_ENABLE_PREVIEW_BUILD is enabled? So the "if" should be ifdef'd but not the call in it? Or is there code elsewhere that takes care of it?

File: src/sst/core/componentInfo.cc [hunk 3]
```diff
@@ -143,12 +143,14 @@ ComponentInfo::ComponentInfo(ComponentInfo &&o) :
 ComponentInfo::~ComponentInfo() {
     if ( link_map ) delete link_map;
     if ( component ) {
+#ifndef SST_ENABLE_PREVIEW_BUILD        
         // For backward compatibility, don't delete component defined
```
Comment (gvoskuilen) [inline src/sst/core/componentInfo.cc]: Similar to the other case, should the 'if' statement be ifdefed but not the contents? So all component defined subcomponents get deleted?

File: src/sst/core/baseComponent.cc (hunk 2 above)
Comment (feldergast) [reply src/sst/core/baseComponent.cc]: Fixed and pushed.

File: src/sst/core/componentInfo.cc (hunk 3 above)
Comment (feldergast) [reply src/sst/core/componentInfo.cc]: Fixed and pushed.

**Pull request merged**